# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/TextEditor.h
# Commit: edac234f1eba
# Full Hash: edac234f1eba144f9da6a10097da779749a9e90f
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-09-21 03:56:08
# Regressor Bug: 1789967
# File Overlap Count: 1
# Description:
#   Bug 1789967 - part 1: Make `TextEditor` and `HTMLEditor` implement `EditorBase::InitEditorContentAndSelection` by themselves r=m_kato
#   
#   The method is enough simple, and uses bad cast from point of view of OOP.
#   Therefore, this patch make the sub classes implement the method only for each.
#   
# ==============================================================================

diff -r 4ec21c88e8c9 -r edac234f1eba editor/libeditor/TextEditor.h
--- a/editor/libeditor/TextEditor.h	Tue Sep 20 23:29:09 2022 +0000
+++ b/editor/libeditor/TextEditor.h	Wed Sep 21 00:20:26 2022 +0000
@@ -534,6 +534,13 @@
   bool CanEchoPasswordNow() const;
 
   /**
+   * InitEditorContentAndSelection() may insert a padding `<br>` element for
+   * if it's required in the anonymous `<div>` element or `<body>` element and
+   * collapse selection at the end if there is no selection ranges.
+   */
+  [[nodiscard]] MOZ_CAN_RUN_SCRIPT nsresult InitEditorContentAndSelection();
+
+  /**
    * Make the given selection span the entire document.
    */
   MOZ_CAN_RUN_SCRIPT nsresult SelectEntireDocument() final;
